python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -193,6 +195,7 @@ def multi_asset(
     description: Optional[str] = None,
     required_resource_keys: Optional[Set[str]] = None,
     compute_kind: Optional[str] = None,
+    internal_asset_deps: Optional[Mapping[str, Set[AssetKey]]] = None,
```

Comment:
Nitpick: prefer AbstractSet to Set

Reply:
AbstractSet doesn't have the .difference() method in the interface, which I use in here. Also, this function should really only be called in a single place, so it seems better to me to clearly define the interface ("yes, we really do expect a set() object").

For similar reasons, I converted the build_asset_ins/build_asset_outs functions to explicitly return Dicts, because they do genuinely return dict objects, and that means we don't have to do that weird { k, v for k, v in mapping.items() } thing.

Reply:
> AbstractSet doesn't have the .difference() method in the interface

I believe you can just do `-`.  I.e. `set1 - set`.

> Also, this function should really only be called in a single place, so it seems better to me to clearly define the interface ("yes, we really do expect a set() object").

An advantage of `AbstractSet` is that it communicates the readers (and enforces) that the function won't mutate the argument. I think that's useful even when there's only a single caller.

That said, not a hill I want to die on at all. Feel free to merge without any changes here.